octopus-server/src/main/java/octopus/teamcity/server/generic/RunbookRunStep.java

```diff
@@ -0,0 +1,72 @@
+package octopus.teamcity.server.generic;
+
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.Lists;
+import com.intellij.openapi.util.text.StringUtil;
+import jetbrains.buildServer.serverSide.InvalidProperty;
+import octopus.teamcity.common.runbookrun.RunbookRunPropertyNames;
+import octopus.teamcity.common.runbookrun.RunbookRunUserData;
+
+public class RunbookRunStep extends OctopusBuildStep {
+
+  private final RunbookRunPropertyNames KEYS = new RunbookRunPropertyNames();
+
+  public RunbookRunStep() {
+    super(
+        "runbook-run",
+        "Runbook run",
+        "editRunbookRunParameters.jsp",
+        "viewRunbookRunParameters.jsp");
+  }
+
+  @Override
+  public String describeParameters(Map<String, String> parameters) {
+    final RunbookRunUserData runbookRunUserData = new RunbookRunUserData(parameters);
+    return String.format(
+        "Runbook name: %s\nProject name: %s",
+        runbookRunUserData.getRunbookName(), runbookRunUserData.getProjectName());
+  }
+
+  @Override
+  public List<InvalidProperty> validateProperties(Map<String, String> properties) {
+    final List<InvalidProperty> failedProperties = Lists.newArrayList();
+
+    if (StringUtil.isEmpty(properties.getOrDefault(KEYS.getRunbookNamePropertyName(), ""))) {
```

Comment:
do we need to do the getOrDefault? I.e. doesn't isEmpty check against null?

Reply:
Fixed